### ISSUE TITLE
Early overfit?

### ISSUE THREAD

**Issue by ifeherva** [dichotomies/proxy-nca#7]
Hi, I have a similar implementation in mxnet but after the first epoch both the test and train accuracy goes down (though loss as well) instead of going up. I tried your code as well with my dataset and the same thing happens. Do proxies overfit after 1 epoch?

**Comment by ifeherva** [comment]
Correct, first I thought it is an overfitting issue then I saw that training accuracy decreases as well. I saw your log file and I can reproduce it on CUB but not on my data (which is not that different, actually larger). Any idea why the learning is not working properly?

**Comment by dichotomies** [comment]
I don't know which dataset you are using, so I can tell you only some common pitfalls.

1. The most important (and trivial) one: optimizer settings. Especially if you are using a different network (for example with VGG you shouldn't choose such a low epsilon).

2. When I tried out the loss for Cars196 (which has 16000 images compared to 6000 of CUB), then a decaying learning rate was important.

3. Make sure you are using data augmentation (see data loader of this repository or the proxy NCA paper).

**Comment by ifeherva** [comment]
Thanks for the suggestions, I am doing all of those. However, I just realized I had a nasty class-inbalance which I removed and now it works a lot better!

**Comment by dichotomies** [comment]
May I ask how you solved your problem of class-imbalance (weighting, skipping, generating samples)?

**Comment by ifeherva** [comment]
I would not say I solved, it just works better :) I am just downsampling currently and experimenting with weighting.
I still experience the initial quick rise then gradual fall of accuracies though (train and test), I think this is a property of the proxies+data. I generate the proxies with uniform distributed random weights so the they are spread evenly which makes good clustering possible, but then due to the nature of my training data they get "pulled too close" together I think. Still investigating :)

**Comment by dichotomies** [comment]
I observed the same behaviour of the proxies while fine-tuning the network for CUB. It was quite frustrating. The training accuracy increased to something like R@1 and NMI > 80%, but the evaluation accuracy decreased steadily at some point.

Probably it's really a matter of this particular loss/proxies. I didn't notice such overfitting when I trained the net with triplet loss and semi-hard sampling.

Good luck with your investigation! :)
